### ISSUE TITLE
[FEATURE] OAuth with i.e. GitLab

### ISSUE THREAD

**Issue by miko007** [polonel/trudesk#134]
**Is this a BUG REPORT or FEATURE REQUEST?**:

- [ ] BUG 
- [X] FEATURE

**Proposal**:
We are currently looking into using trudesk as a lightweight replacement for OTRS. We are highly dependent of an OAuth 2 login functionality, but we could not find too much info about truedesk on that matter.

* Would it be possibile to write a plugin which provided this functionality?
* Would the maintainers/the community also be interested in this feature and would a pull request implementing it be accepted in general?

Thanks already for the awesome work you put in.

**Comment by polonel** [comment]
The way plugins are implemented right now it wouldn't be possible to have a plugin at this time. However, implementing OAuth 2 is something that I want the project to support generally. It was planned to be part of the API overhaul which would support a configurable login strategy. Trudesk uses [Passport.js](http://www.passportjs.org) for its auth, which would make it easy to configure which strategy to use. It would require configuring the login strategy during initial install. 

I'm open to pull requests and discussion on general implementation. It is indeed a feature I want the project to have.

**Comment by miko007** [comment]
i do not think this issue should be stalled, as we need definately further discussion here.

The Implementation has to be thoroughly thought through, so it meets nicely with all the option which are there already. Also the location of the options screen etc. must be discussed, until further development on this end.

**Comment by polonel** [comment]
I have the marked the issue so stalebot will leave it be.

**Comment by miko007** [comment]
the current status here is, that the backend ui code is nearly impossible to maintain. the files are way too big and the views are basically a mess. this makes it really hard to start implementing new features, because something breaks pretty easy while trying to add a new settings page for configuration.

are there plans to refactor that part of the code, yet?

**Comment by miko007** [comment]
never mind, i did not merge upstream in a while and there have happened major improvements. ic now will take a closer look again and implement some stuff.

**Comment by polonel** [comment]
I agree the codebase was a very big mess using Angular and the files grew largely. The decision to drop angular was made in February and since the codebase has been moving to react in stages. In its current state, it's running both angular and react as the components move. In the end, angular will be dropped and the UI will be made up of React/Redux. 

Sorry for the mess right now but it is on a path to better things.

**Comment by geeeksameer** [comment]
Any update on OAuth integration, @polonel sir?

**Comment by polonel** [comment]
@geeeksameer This is a feature that was implemented in the Cloud version of Trudesk. I'll see if I'm able to port it over to the community edition.

**Comment by geeeksameer** [comment]
It would be fabulous, truly.
As, I have been working to deploy this amazing open source helpdesk solution for my app, I've configured this whole system on my servers and looked closely to understand the whole system flow, customized it a bit according to our needs.
The last thing that I desperately want to be configured is OAuth, I would be really thankful if you help with that and would sponsor you for sure.

**Comment by polonel** [comment]
@geeeksameer What provider are you looking to integrate? As this was a feature of the cloud version, I may be able to implement a select few.

- Github
- Gitlab
- Facebook
- Google
- Microsoft
- Other

**Comment by majikman111** [comment]
@polonel My organization would also greatly appreciate if this feature was ported to community edition. For reference, we would want to use GitLab

**Comment by orobourosfkt** [comment]
@polonel I'm alsow interested in this feature (Microsoft and/or Gitlab) :-)

**Comment by sfulham** [comment]
It would also be nice if you were to implement SAML as well!

**Comment by sfulham** [comment]
Any updates?

**Comment by jgardner-qha** [comment]
Would LOVE this! Either SAML or OAuth2 for Microsoft integration would be a fantastic feature.

**Comment by dc-dp** [comment]
We're using Authentik for OIDC/OAuth2/SAML support for most of our apps, so a generic OIDC connector would be huge for us. Thanks for looking into this :)

**Comment by MyWay** [comment]
+1

**Comment by polonel** [comment]
This is included in the Cloud V2 Planned list.
